.github/workflows/_openxla_pin_update_weekly.yml

```diff
@@ -0,0 +1,52 @@
+name: Weekly Code Update PR
+on:
+  schedule:
+    - cron: '0 9 * * 1'  # Each Monday 9:00AM (PDT time)
+
+jobs:
+  update_and_create_pr:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Get current date
+        run: echo "CURRENT_DATE=$(date +'%Y%m%d')" >> $GITHUB_ENV
+
+      - name: Install sed
+        run: sudo apt-get update && sudo apt-get install -y sed
+
+      - name: Update date in setup.py
+        run: |
+          sed -i "s/_date =.*/_date = '${{ env.CURRENT_DATE }}'/" ${FILE_PATH}
+          git add ${FILE_PATH}
+          git commit -m "Weekly update setup.py to (${{ env.CURRENT_DATE }})"
+          git push origin HEAD:weekly-update
```

Comment:
Can you create a new branch instead for each update? Currently this will fail on the second push, because `weekly-update` will fall out of date with `master`. And you don't want to force-push over the old branch if that PR is still open.

Reply:
Thanks, make sense, updated